qiskit_experiments/library/characterization/analysis/resonator_spectroscopy_analysis.py

```diff
@@ -0,0 +1,72 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy analysis class for resonators."""
+
+from typing import List, Tuple
+import numpy as np
+
+import qiskit_experiments.curve_analysis as curve
+from qiskit_experiments.curve_analysis import ResonanceAnalysis
+from qiskit_experiments.framework import AnalysisResultData, ExperimentData
+from qiskit_experiments.framework.matplotlib import get_non_gui_ax
+from qiskit_experiments.data_processing.nodes import ProjectorType
+
+
+class ResonatorSpectroscopyAnalysis(ResonanceAnalysis):
+    """Class to analysis resonator spectroscopy."""
+
+    @classmethod
+    def _default_options(cls):
+        options = super()._default_options()
+        options.dimensionality_reduction = ProjectorType.ABS
+        options.result_parameters = [curve.ParameterRepr("freq", "meas_freq", "Hz")]
```

Comment:
#171 had proposed adding non-qubit components and flagged this experiment as likely the first experiment to add such a component. Do we want to set the component for the result to be a `Resonator` instead of a `Qubit`? This is what the current production system uses.

Reply:
That makes sense. It does not seem straight forward to implement without a small change to `BaseAnalysis`. How about this: 9dbe059?

Reply:
That looks good to me. It is preferrable to the current patch in #387 because it allows making the resonator the only component whereas #387 would result in a qubit and resonator component (unless you unset `physical_qubits` which could have some side effects). We might later need to add more flexibility to set components per result. With our older code, we have an experiment like the one in this PR that does resonator spectroscopy but with and without a pi on the qubit. In that case, it extracts resonator frequency and linewidth results with the resonator component set and the qubit-resonator chi with both the qubit and resonator components set.

Should we propose this change in #387 since a different set of reviewers were interested there?

Reply:
That makes sense to me. We can propose this change in #387. Alternatively we simply merge this PR and close #387.